### PR TITLE
Process API

### PR DESCRIPTION
**Pull request by DiMino-0**
This api will facilitate interactions between the DSS and CE. Generic types are interfaces, prototype class to make sure it compiles.

### CODE REVIEW

File: src/ProcessApi/ComputeEngine.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/ComputeEngine.java]: change the package name to processapi. the naming convention in check style rules it

File: src/ProcessApi/PrototypeProcess.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package ProcessApi;
+
+public class PrototypeProcess {
+	public void prototype(ComputeEngine ce, DataStorage ds)
+	{
```
Comment (RodolfoBaez) [inline src/ProcessApi/PrototypeProcess.java]: move this '{' back to line 4. breaks checkstyle_check.xml rule.

File: src/ProcessApi/DSCode.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+public enum DSCode {
+	SUCCESS(1, true),
+	FALIURE(0, false);
+	
+	private final int id;
+	private final boolean success;
+	
+	private DSCode(int id, boolean success)
+	{
```
Comment (RodolfoBaez) [inline src/ProcessApi/DSCode.java]: can you move the '{' on line 11 to line 10.

File: src/ProcessApi/DSCode.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/DSCode.java]: change the package name to processapi. the naming convention in check style rules it

File: src/ProcessApi/DataStorage.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/DataStorage.java]: change the package name to processapi. the naming convention in check style rules it

File: src/ProcessApi/InputSource.java [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/InputSource.java]: change the package name to processapi. the naming convention in check style rules it

File: src/ProcessApi/OutputSource.java [hunk 7]
```diff
@@ -0,0 +1,5 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/OutputSource.java]: change the package name to processapi. the naming convention in check style rules it

File: src/ProcessApi/ProcessInterface.java [hunk 8]
```diff
@@ -0,0 +1,6 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/ProcessInterface.java]: change the package name to processapi. the naming convention in check style rules it

File: src/ProcessApi/PrototypeProcess.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
+package ProcessApi;
```
Comment (RodolfoBaez) [inline src/ProcessApi/PrototypeProcess.java]: change the package name to processapi. the naming convention in check style rules it

**Review by RodolfoBaez — CHANGES_REQUESTED** [review]
hey Joseph, can you take a look at the comments, it should allow the checkstyle_check.xml approve changes for commit to proceed. Thanks.

File: src/processAPI/PrototypeProcess.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+	public void prototype(ComputeEngine CE, DataStorage DS)
+	{
+		//get input src from CE
+		InputSource input = CE.getInputSrc();
+		//get output src from CE
+		OutputSource output = CE.getOutputSrc();
+		//write to Output
+		DSCode outputCode = DS.writeToOutput();
+		//read from Input
+		DSCode inputCode = DS.readFromInput();
```
Comment (LourisA11) [inline src/processAPI/PrototypeProcess.java]: this needs a parameter so it knows it knows where it read from. (input)

**Pull request merged**